examples/sampling/node_classification.py

```diff
@@ -0,0 +1,240 @@
+import torch.nn as nn
+import torch.nn.functional as F
+import torchmetrics.functional as MF
+import tqdm
+from dgl.data import AsNodePredDataset
+from dgl.dataloading import (
+    DataLoader,
+    MultiLayerFullNeighborSampler,
+    NeighborSampler,
+)
+from ogb.nodeproppred import DglNodePropPredDataset
+
+
+class SAGE(nn.Module):
+    def __init__(self, in_size, hid_size, out_size):
+        super().__init__()
+        self.layers = nn.ModuleList()
+        # Three-layer GraphSAGE-mean.
+        self.layers.append(dglnn.SAGEConv(in_size, hid_size, "mean"))
+        self.layers.append(dglnn.SAGEConv(hid_size, hid_size, "mean"))
+        self.layers.append(dglnn.SAGEConv(hid_size, out_size, "mean"))
+        self.dropout = nn.Dropout(0.5)
+        self.hid_size = hid_size
+        self.out_size = out_size
+
+    def forward(self, blocks, x):
+        h = x
+        for layer_idx, (layer, block) in enumerate(zip(self.layers, blocks)):
+            h = layer(block, h)
+            is_last_layer = layer_idx == len(self.layers) - 1
+            if not is_last_layer:
+                h = F.relu(h)
+                h = self.dropout(h)
+        return h
+
+    def inference(self, g, device, batch_size):
+        """Conduct layer-wise inference to get all the node embeddings."""
+        feat = g.ndata["feat"]
+        #####################################################################
+        # (HIGHLIGHT) Creating a MultiLayerFullNeighborSampler instance.
+        # This sampler is used in the Graph Neural Networks (GNN) training
+        # process to provide neighbor sampling, which is crucial for
+        # efficient training of GNN on large graphs.
+        #
+        # The first argument '1' indicates the number of layers for
+        # the neighbor sampling. In this case, it's set to 1, meaning
+        # only the direct neighbors of each node will be included in the
+        # sampling.
+        #
+        # The 'prefetch_node_feats' parameter specifies the node features
+        # that need to be pre-fetched during sampling. In this case, the
+        # feature named 'feat' will be pre-fetched, which should be part
+        # of the nodes' data.
+        #####################################################################
+        sampler = MultiLayerFullNeighborSampler(1, prefetch_node_feats=["feat"])
+
+        dataloader = DataLoader(
+            g,
+            torch.arange(g.num_nodes()).to(g.device),
+            sampler,
+            device=device,
+            batch_size=batch_size,
+            shuffle=False,
+            drop_last=False,
+            num_workers=0,
+        )
+        buffer_device = torch.device("cpu")
+        pin_memory = buffer_device != device
+
+        for layer_idx, layer in enumerate(self.layers):
+            is_last_layer = layer_idx == len(self.layers) - 1
+            y = torch.empty(
+                g.num_nodes(),
+                self.hid_size if (not is_last_layer) else self.out_size,
```

Comment:
```suggestion
                self.out_size if is_last_layer else self.hid_size,
```
I tend to find this more readable.